src/com/amazon/kinesis/streaming/agent/tailing/SourceFile.java

```diff
@@ -114,6 +115,13 @@ public String toString() {
     private void validateDirectory(Path dir) {
         Preconditions.checkArgument(dir != null, "Directory component is empty!");
         // TODO: validate that the directory component has no glob characters
+
+        //check the permissions of every level in the directory path
+        for(Path level : dir.toAbsolutePath()){
+            if(!Files.isReadable(level)){
```

Comment:
Since there is a requirement that every directory up to the file needs to be readable by the agent user, this checks whether that is the case. If it isn't it throws a warning.